linker.go

```diff
@@ -232,32 +265,29 @@ fixups:
 		// check meta, if no meta return err
 		kfm, _ := ins.Metadata.Get(kfuncMeta{}).(*btf.Func)
 		if kfm == nil {
-			return fmt.Errorf("kfunc call has no kfuncMeta")
-		}
-
-		var fn *btf.Func
-		if err := s.TypeByName(kfm.Name, &fn); err != nil {
-			return fmt.Errorf("couldn't resolve %s in kernel spec: %v: %w", kfm.Name, err, ErrNotSupported)
+			return nil, fmt.Errorf("kfunc call has no kfuncMeta")
 		}
 
-		if err := btf.CheckTypeCompatibility(kfm.Type, fn.Type); err != nil {
-			return err
+		idx := 0
```

Comment:
Nit: is it necessary to declare `idx` here?

Reply:
Fixed :ballot_box_with_check: